packages/features/bookings/lib/handleSeats.ts

```diff
@@ -0,0 +1,796 @@
+        const credential = await prisma.credential.findUnique({
+          where: {
+            id: reference.credentialId,
+          },
+          select: credentialForCalendarServiceSelect,
+        });
+
+        if (credential) {
+          if (reference.type.includes("_video")) {
+            integrationsToDelete.push(deleteMeeting(credential, reference.uid));
+          }
+          if (reference.type.includes("_calendar") && originalBookingEvt) {
+            const calendar = await getCalendar(credential);
+            if (calendar) {
+              integrationsToDelete.push(
+                calendar?.deleteEvent(reference.uid, originalBookingEvt, reference.externalCalendarId)
+              );
+            }
+          }
+        }
+      }
+    }
+
+    await Promise.all(integrationsToDelete).then(async () => {
+      await prisma.booking.update({
+        where: {
+          id: originalRescheduledBooking.id,
+        },
+        data: {
+          status: BookingStatus.CANCELLED,
+        },
+      });
+    });
+    deletedReferences = true;
+  }
+  return deletedReferences;
+};
+
+const handleSeats = async ({
+  rescheduleUid,
+  reqBookingUid,
+  eventType,
+  evt,
+  invitee,
+  allCredentials,
+  organizerUser,
+  originalRescheduledBooking,
+  bookerEmail,
+  tAttendees,
+  bookingSeat,
+  reqUserId,
+  rescheduleReason,
+  reqBodyUser,
+  noEmail,
+  isConfirmedByDefault,
+  additionalNotes,
+  reqAppsStatus,
+  attendeeLanguage,
+  paymentAppData,
+  fullName,
+  smsReminderNumber,
+  eventTypeInfo,
+  uid,
+  eventTypeId,
+  reqBodyMetadata,
+  subscriberOptions,
+  eventTrigger,
+}: {
+  rescheduleUid: string;
+  reqBookingUid: string;
+  eventType: NewBookingEventType;
+  evt: CalendarEvent;
+  invitee: Invitee;
+  allCredentials: Awaited<ReturnType<typeof getAllCredentials>>;
+  organizerUser: OrganizerUser;
+  originalRescheduledBooking: OriginalRescheduledBooking;
+  bookerEmail: string;
+  tAttendees: TFunction;
+  bookingSeat: BookingSeat;
+  reqUserId: number | undefined;
+  rescheduleReason: RescheduleReason;
+  reqBodyUser: string | string[] | undefined;
+  noEmail: NoEmail;
+  isConfirmedByDefault: IsConfirmedByDefault;
+  additionalNotes: AdditionalNotes;
+  reqAppsStatus: ReqAppsStatus;
+  attendeeLanguage: string | null;
+  paymentAppData: PaymentAppData;
+  fullName: ReturnType<typeof getFullName>;
+  smsReminderNumber: SmsReminderNumber;
+  eventTypeInfo: EventTypeInfo;
+  uid: short.SUUID;
+  eventTypeId: EventTypeId;
+  reqBodyMetadata: ReqBodyMetadata;
+  subscriberOptions: GetSubscriberOptions;
+  eventTrigger: WebhookTriggerEvents;
+}) => {
+  const loggerWithEventDetails = createLoggerWithEventDetails(eventType.id, reqBodyUser, eventType.slug);
+
+  let resultBooking:
+    | (Partial<Booking> & {
+        appsStatus?: AppsStatus[];
+        seatReferenceUid?: string;
+        paymentUid?: string;
+        message?: string;
+        paymentId?: number;
+      })
+    | null = null;
```

Comment:
Could we assign a proper type to this and re-use it across the file? Rather than enforcing a type on the function call.

Reply:
This is handled in #12905